lib/app.js

```diff
@@ -80,9 +80,9 @@ App.bootstrap = function (app, root) {
   app.rootDir = root = root || app.rootDir;
 
   app.paths = {
-    preboots: app.given.preboots || path.join(root, 'lib', 'preboots'),
-    middlewares: app.given.middlewares || path.join(root, 'lib', 'middlewares'),
-    routes: app.given.routes || path.join(root, 'lib', 'routes')
+    preboots: typeof app.given.preboots === 'string' ? app.given.preboots : path.join(root, 'lib', 'preboots'),
+    middlewares: typeof app.given.middlewares === 'string' ? app.given.middlewares : path.join(root, 'lib', 'middlewares'),
+    routes: typeof app.given.routes === 'string' ? app.given.routes : path.join(root, 'lib', 'routes')
```

Comment:
Semantically is this the behavior we really want? In the case any of these are functions falling back to the default value is setting a default that doesn't represent the actual location of the logic. 

In other words: what's the use of supplying a default value if it is `ENOENT`?

Reply:
Would it be acceptable to leave any that are provided as functions `undefined` in `paths`?

Reply:
That appears to be the correct approach from my perspective. At least this way we are not lying to the user w.r.t. files that exist on disk. e.g.:

``` js
app.paths = ['preboots', 'middlewares', 'routes'].reduce((acc, name) => {
  if (typeof app.given[name] === 'string') {
    acc[name] = app.given[name];
  }

  return acc; 
}, {});
```